### PR TITLE
Add UI for deleting text annotations

### PR DESCRIPTION
**Pull request by tedlee**
This PR implements the following:
- Refactors TextColorModeButton to use updated assets
- The cursor color of JotTextEditView now matches the text color
- Adds a trash button to the editing annotation screen
- Adds a "Done" button to the editing annotation screen

### Screenshot
simulator screen shot - iphone 8 - 2018-01-10 at 16 22 21

### CODE REVIEW

File: jot/JotTextEditView.m [hunk 1]
```diff
@@ -123,6 +123,45 @@ - (instancetype)init
             make.width.equalTo(@32);
             make.height.equalTo(@20);
         }];
+
+        NSBundle *bundle = [NSBundle bundleForClass:[self class]];
+        NSURL *url = [bundle URLForResource:@"jot" withExtension:@"bundle"];
+        
+        self.trashcanButton = [[UIButton alloc] init];
+        [self.trashcanButton setImage:[UIImage imageNamed:@"trashcan" inBundle:[NSBundle bundleWithURL:url] compatibleWithTraitCollection:nil] forState:UIControlStateNormal];
```
Comment (hellomayuko) [inline jot/JotTextEditView.m:131]: woah, interesting way to import an image. Does it not just work with the Asset catalog?

File: jot/JotTextEditView.m [hunk 2]
```diff
@@ -123,6 +123,45 @@ - (instancetype)init
...
+        [self.trashcanButton addTarget:self action:@selector(deleteAnnotation) forControlEvents:UIControlEventTouchUpInside];
+        self.trashcanButton.hidden = YES;
+        [self addSubview:self.trashcanButton];
+
+        [self.trashcanButton mas_makeConstraints:^(MASConstraintMaker *make) {
```
Comment (hellomayuko) [inline jot/JotTextEditView.m:136]: woah what's this `mas_makeConstraints` thing?

File: jot/JotTextEditView.m [hunk 3]
```diff
@@ -123,6 +123,45 @@ - (instancetype)init
...
+            // Respect safeArea on iOS11
+            if (@available(iOS 11.0, *)) {
+                make.top.equalTo(self.mas_safeAreaLayoutGuideTop).offset(24.f);
```
Comment (hellomayuko) [inline jot/JotTextEditView.m:139]: 🙌

File: jot/JotTextEditView.m [hunk 4]
```diff
@@ -23,8 +23,10 @@ @interface JotTextEditView () <UITextViewDelegate>
 @property (nonatomic, strong) CAGradientLayer *gradientMask;
 @property (nonatomic, strong) CAGradientLayer *topGradient;
 @property (nonatomic, strong) CAGradientLayer *bottomGradient;
-@property (nonatomic, strong) TextColorModeButton *backgroundColorMode;
+@property (nonatomic, strong) UIButton *trashcanButton;
+@property (nonatomic, strong) UIButton *textAnnotationDoneButton;
 @property (nonatomic, strong) TextAlignmentButton *textAlignmentButton;
+@property (nonatomic, strong) TextColorModeButton *backgroundColorMode;
```
Comment (hellomayuko) [inline jot/JotTextEditView.m]: maybe this property should also be postfixed with `button`?

**Review by hellomayuko — APPROVED** [review]
Small nits and questions, overall looks good.

File: jot/JotTextEditView.m (hunk 1 above)
Comment (tedlee) [reply jot/JotTextEditView.m:131]: Yeah Cocapods recommends defining a resource bundle that's defined in the `podspec`: http://guides.cocoapods.org/syntax/podspec.html#resources

**Pull request merged**